neonvm/apis/neonvm/v1/virtualmachine_types.go

```diff
@@ -196,9 +196,60 @@ type GuestSettings struct {
 	// +optional
 	Sysctl []string `json:"sysctl,omitempty"`
 
-	// Swap adds a swap disk with the provided size.
+	// Swap controls settings for adding a swap disk to the VM.
 	// +optional
-	Swap *resource.Quantity `json:"swap,omitempty"`
+	Swap *SwapInfo `json:"swap,omitempty"`
+}
+
+type SwapInfo struct {
+	// Size sets the size of the swap disk on the host. The amount available within the VM may be
+	// slightly less (typically: 4KiB less, or 3MiB+4KiB less if Shrinkable = true).
+	Size resource.Quantity `json:"size"`
+	// Shrinkable changes the swap to sit inside a GPT partition of the mounted disk, so that it can
+	// be resized from within the guest.
+	//
+	// Resizing can be done with the provided program at '/neonvm/bin/resize-swap'. It requires
+	// superuser permissions to perform the resizing. Please note that this DISABLES ALL SWAP before
+	// enabling it at the new size.
+	//
+	// +optional
+	Shrinkable *bool `json:"shrinkable,omitempty"`
```

Comment:
Is there a real need to make it not shrinkable?

Reply:
Made it always shrinkable with e77479e9f055aaa0f0a196d48e86be9cd0040aa3